server/src/main/java/module-info.java

```diff
@@ -384,6 +384,7 @@
     uses org.elasticsearch.jdk.ModuleQualifiedExportsService;
     uses org.elasticsearch.node.internal.TerminationHandlerProvider;
     uses org.elasticsearch.internal.VersionExtension;
+    uses org.elasticsearch.common.settings.PublicSettingsFactory;
```

Comment:
if loading via LoaderExtension we don't need this (along with the provides on the plugin side)
leaving comment for discussion: loaderExtension vs spi